src/main/java/dev/lukebemish/ziggradle/BaseZigOptions.java

```diff
@@ -0,0 +1,88 @@
+package dev.lukebemish.ziggradle;
+
+import org.gradle.api.Named;
+import org.gradle.api.file.DirectoryProperty;
+import org.gradle.api.invocation.Gradle;
+import org.gradle.api.model.ObjectFactory;
+import org.gradle.api.provider.ListProperty;
+import org.gradle.api.provider.MapProperty;
+import org.gradle.api.provider.Property;
+import org.gradle.api.tasks.Input;
+import org.gradle.api.tasks.Internal;
+import org.gradle.api.tasks.Optional;
+
+import javax.inject.Inject;
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.List;
+
+public abstract class BaseZigOptions {
+
+    @Input
+    public abstract MapProperty<String, String> getEnv();
+
+    @Input
+    @Optional
+    public abstract Property<Boolean> getReplaceEnv();
+
+    @Input
+    public abstract ListProperty<Object> getCompilerArgs();
```

Comment:
This completely tosses out any type safety and there is no reason for this to be anything except a `ListProperty<String>`